structure/stack/stackarray.go

```diff
@@ -9,31 +9,45 @@
 
 package stack
 
-var stackArray []any
+type StackArray struct {
+	elements []interface{}
```

Comment:
Use a generic type parameter instead of `any` (`interface{}`).

Reply:
`
  type Array[T any] struct {
	  elements []T
  }
`
Will this be fine or should I create a separate struct for the allowed data types ?

Reply:
A Stack Array should support any data type correct? So you should use `any`.

Reply:
I think that the usage of `type Array[T any] struct { elements []T }` might be better. Consider this example:

```go
...
import "fmt"

type Array[T any] struct{ elements []T }

func main() {
	data := Array[any]{
		elements: []any{1, "a"},
	}
	fmt.Println(data)
}
...
```

Here we have the same usage as the current implementation with the added benefit that the user should be able to make the `Stack` a typed stack. Though overall I don't know whether any one way is better or worse than the other... 😄

Reply:
@tjgurwara99 Thats what they are implementing rn? Not sure how your suggestion is different.

Reply:
> @tjgurwara99 Thats what they are implementing rn? Not sure how your suggestion is different.

The only extra benefit for using the approach mentioned is that the user would have to choice to make the this stack a "typed" stack or keep the stack dynamic in nature.

For example

```
...
import "fmt"

type Array[T any] struct{ elements []T }

func Example() {
	data := Array[any]{
		elements: []any{1, "a"},
	}
	fmt.Printf("%T\n", data.elements)
	data2 := Array[int]{
		elements: []int{1, 2, 3},
	}
	fmt.Printf("%T\n", data2.elements)
	// Output:
	// []interface {}
	// []int
}
...
```

You can imagine that if a user only needs to use the `int` type inside the stack, then they won't need to do type casting and type checks like `d, ok := data.elements[0].(int)` all the time... Though arguably it is indeed a contrived example, so not sure how useful it would be out in the wild.

Reply:
Yeah but thats what we discussed about implementing right? Is your suggestion any different from:

> Use a generic type parameter instead of `any` (`interface{}`).

> `type Array[T any] struct { elements []T }`
> Will this be fine or should I create a separate struct for the allowed data types ?

Reply:
> Use a generic type parameter instead of `any` (`interface{}`).

Probably not, I guess there is some miscommunication here. I assumed when you mentioned the above, you were talking about just replacing the `interface{}` to `any`. And I got further confused by the following where I assumed that you wanted to keep the older implementation rather than what the author suggested...

> A Stack Array should support any data type correct? So you should use any.

Anyways, I think we're both on the same page which means my suggestion was no different from what was discussed 😄

Reply:
Yeah seems the issue has been resolved :)